bundles/core/src/main/java/com/adobe/cq/wcm/core/components/models/Download.java

```diff
@@ -0,0 +1,217 @@
+ ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ ~ See the License for the specific language governing permissions and
+ ~ limitations under the License.
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~*/
+
+package com.adobe.cq.wcm.core.components.models;
+
+import com.adobe.cq.export.json.ComponentExporter;
+
+import javax.annotation.Nonnull;
+
+/**
+ * Defines the {@code Download} Sling Model for the {@code /apps/core/wcm/components/download} component.
+ */
+public interface Download extends ComponentExporter {
+
+    /**
+     * Name of the resource property that defines whether or not the title value is taken from the configured asset.
+     */
+    String PN_TITLE_FROM_ASSET = "titleFromAsset";
+
+    /**
+     * Name of the resource property that defines whether or not the description value is taken from the configured asset.
+     */
+    String PN_DESCRIPTION_FROM_ASSET = "descriptionFromAsset";
+
+    /**
+     * Name of the resource property that defines whether or not the download item should be displayed inline vs. attachment.
+     */
+    String PN_INLINE = "inline";
+
+
+    /**
+     * Name of the policy property that defines the text to be displayed on the Call-to-Action.
+     */
+    String PN_ACTION_TEXT = "actionText";
+
+    /**
+     * Name of the policy property that stores the value for this title's HTML element type.
+     *
+     * @see #getTitleType()
+     */
+    String PN_TITLE_TYPE = "titleType";
+
+    /**
+     * Name of the policy property that defines whether an image representing the file will be displayed.
+     */
+    String PN_DISPLAY_THUMBNAIL = "displayImage";
+
+    /**
+     * Name of the policy property that defines whether the file's size will be displayed.
+     */
+    String PN_DISPLAY_SIZE = "displaySize";
+
+    /**
+     * Name of the policy property that defines whether the file's format will be displayed.
+     */
+    String PN_DISPLAY_FORMAT = "displayFormat";
+
+    /**
+     * Name of the policy property that defines whether the filename will be displayed.
+     */
+    String PN_DISPLAY_FILENAME = "displayFilename";
+
+
+    /**
+     * Returns either the title configured in the dialog or the title of the DAM asset,
+     * depending on the state of the titleFromAsset checkbox.
+     *
+     * @return the download title
+     */
+    default String getTitle() {
+        throw new UnsupportedOperationException();
+    }
+
+    /**
+     * Returns either the description configured in the dialog or the description of the DAM asset,
+     * depending on the state of the descriptionFromAsset checkbox.
+     *
+     * @return the download description
+     */
+    default String getDescription() {
+        throw new UnsupportedOperationException();
+    }
+
+    /**
+     * Returns the url to the asset.
+     *
+     * @return the asset url
+     */
+    default String getDownloadUrl() {
```

Comment:
`Download` seems redundant here, perhaps just `getUrl` or `getLink` (as we commonly use in other components).